test/Engine/ProtoTest/Associative/FunctionObject.cs

```diff
@@ -594,5 +594,94 @@ public void Test__GroupByKey4()
             thisTest.Verify("r1", new object[] { new object[] { "item1" } });
             thisTest.Verify("r2", new object[] { "key1", "key2" });
         }
+
+        [Test]
+        public void Test__SortByKey1()
+        {
+            string code =
+    @"
+import (""DSCoreNodes.dll"");
+import (""FunctionObject.ds"");
+
+list = {""item1"", ""item2""};
+keys = {""key2"", ""key1""};
+
+result = __SortByKey(list, keys);
+r1 = result[0];
+r2 = result[1];
+";
+            thisTest.RunScriptSource(code);
+            thisTest.Verify("r1", new object[] { "item2", "item1" });
+            thisTest.Verify("r2", new object[] { "key1", "key2" });
+        }
+
+        [Test]
+        public void Test__SortByKey2()
+        {
+            string code =
+    @"
+import (""DSCoreNodes.dll"");
+import (""FunctionObject.ds"");
+
+list = {""item1"", ""item2""};
+keys = {""key1""};
+
+result = __SortByKey(list, keys);
+r1 = result[0];
+r2 = result[1];
+";
+            thisTest.RunScriptSource(code);
+            thisTest.Verify("r1", null);
+            thisTest.Verify("r2", null);
+        }
+
+        [Test]
+        public void Test__SortByKey3()
+        {
+            string code =
+    @"
+import (""DSCoreNodes.dll"");
+import (""FunctionObject.ds"");
+
+list = {""item1""};
+keys = {""key1"", ""key2""};
+
```

Comment:
can you add a test where the keys are lists and a ping @lukechurch to see if this should work or not.

ie: keys = {{key1},key2,{key1}}.